sql/handler.h

```diff
@@ -3624,7 +3635,8 @@ const char *ha_legacy_type_name(legacy_db_type legacy_type);
 const char *get_canonical_filename(handler *file, const char *path,
                                    char *tmp_path);
 bool mysql_xa_recover(THD *thd);
-
+bool can_hold_read_locks_on_select(THD *thd, thr_lock_type lock_type);
```

Comment:
These two actually look like unnecessary refactors and could/should probably just have their logic expanded inline. They are used from one place in handler.cc, the new method is_using_prohibited_gap_locks, and was a factor out of ha_innodb.cc.